core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java

```diff
@@ -252,6 +254,310 @@ public void testPlanTasks() {
     validateCombinedScanTasks(tasks, 0);
   }
 
+  @Test
+  public void testPartitionSpecEvolutionRemovalV1() {
+    Assume.assumeTrue(formatVersion == 1);
+
+    // Change spec and add two data and delete files each
+    table.updateSpec().removeField(Expressions.bucket("data", 16))
+        .addField("id").commit();
+    PartitionSpec newSpec = table.spec();
+
+    // Add two data files and two delete files with new spec
+    PartitionKey data10Key = new PartitionKey(newSpec, table.schema());
```

Comment:
An alternative way to define partitioning is to use `withPartitionPath` and specify a string.
I don't have a preference, just FYI.

Reply:
Yea I couldn't get it to work due to the void transform being there, didn't try that hard though.